File: RunLengthEncode.py

# Online Python - IDE, Editor, Compiler, Interpreter
def encode(inp):
    result = ""
    length = len(inp)
    for i in range(length):
        count = 1
        while (i < length - 1 and inp[i] == inp[i + 1]):
            count += 1
            i += 1
        if count > 4:
            result += "/" + str(count).zfill(2) + inp[i]
        else:
            result += inp[i] * count
    return result

input_str = input("Enter between 0 and 50 characters: ")
encoded_str = encode(input_str)
print(encoded_str)

File: RunLengthEncode.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package runlengthencode;
import java.util.Scanner;

/**
 *
 * @author johnc
 */
public class RunLengthEncode 
{

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) 
    {
        // TODO code application logic here
        //String result;
        Scanner keyboard = new Scanner(System.in);
        System.out.println("Enter between 0 and 50 characters");
        
        String input = keyboard.nextLine();
        String value = encode(input);
        System.out.println(value);
        //System.out.println(result);
    }
    
    public static String encode(String in)
    {
        int len = in.length();
        String result = "";
        String twoDigit;
        
        for (int index = 0; index < len; index++)
        {
            int count = 1;
            while (index < len - 1 && in.charAt(index) == in.charAt(index + 1))
            {
                count++;
                index++;
            }
            
            if (count > 4)
            {
                twoDigit = String.format("%02d", count);
                result = result + "/" + twoDigit + in.charAt(index);
            }
            else
            {
                for (int k = 0; k < count; k++)
                {
                    result = result + in.charAt(index);
                }
            }
        }
        return result;
    }
}
